Feb 2026 newsletter (#590)
maplibre/maplibre.github.io: src/content/news/2026-03-03-maplibre-newsletter-february-2026/index.mdx

---
title: "MapLibre Newsletter February 2026"
date: "2026-03-03"
categories: ["newsletter"]
authors:
  [
    harel-mazor,
    bart-louwers,
    frank-elsinga,
    ramya-ragupathy,
    joscha-eckert,
    stephanie-may,
  ]
draft: false
---

import { Image } from "astro:assets";
import mapTilerLogo from "/public/img/maptiler-logo.svg";
import maplibreLogo from "/public/img/maplibre-logos/maplibre-logo-for-light-bg.svg";
import flutteWindows from "./flutter-windows.png";
import maplibreRsText from "./maplibre-rs-text.png";
import qgis2VectorTiles from "./qgis2-vector-tiles.gif";
import mltTui from "./mlt-tui.png";
import mltQgis from "./mlt-qgis.png";

> Welcome to the February 2026 edition of MapLibre Newsletter! This month focuses on cross-platform innovation, from shifting rendering backends to launching AI-ready knowledge libraries.

Author: That's my perception based on what's the backend rendering support mentioned in native and flutter. If that does not sound right I will see how I can rephrase this.

> Welcome to the February 2026 edition of MapLibre Newsletter! This month focuses on cross-platform innovation, from shifting rendering backends to launching AI-ready knowledge libraries.

Author: `preforming` - sorry I don't understand. Did you mean performing?

> A special thanks to MapTiler for their continued support, which is vital to the ongoing development work.

Reviewer: ```suggestion
We updated our [Sponsors](https://maplibre.org/sponsors/) page this month, to make clearer how organizations and individuals can give. If you have been thinking about contributing, there has never been a better time.

A special thanks to YellowMap for becoming a new sponsor, and MapTiler for their continued support.
```

Author: I will commit this suggestion, retain the para on sponsor page update and remove the YellowMap line as it will be a separate announcement post.

The primary change in version 6 regarding developer experience will be the transition to ESM. As the industry moves toward ESM and our current build process is becoming increasingly complex, we are considering moving away from releasing a CommonJS version. This transition will allow us to leverage newer technologies like Rolldown.

We woudl love to hear your thoughts on these expected breaking changes to ensure we don't disrupt your workflows too much.
Please join the discussion in the [relevant tracking issue](https://github.com/maplibre/maplibre-gl-js/issues/6427) and share your feedback.

## 🪶 MapLibre Flutter

Versions [0.3.4](https://github.com/josxha/flutter-maplibre/releases/tag/v0.3.4) and [0.3.4+1](https://github.com/josxha/flutter-maplibre/releases/tag/v0.3.4%2B1) of [maplibre](https://pub.dev/packages/maplibre) have been released this month, bringing significant architectural improvements and expanded platform support.

**iOS transition to FFI**

> A major milestone in this release is the complete transition of the iOS implementation to FFI, removing all remaining usage of method channel serialization. This simplifies the integration between Flutter and the native SDK while providing minor performance gains.

Reviewer 2: ```suggestion
A major milestone in this release is the complete transition of the iOS implementation to use [FFI](https://dart.dev/interop/c-interop), removing all remaining usages of method channel serialization. This simplifies the integration between Flutter and the native SDK while providing minor performance gains.
```

cc @josxha

**Federated plugin architecture**

> As part of an ongoing committment to code quality, the plugin has been reorganized into a fully federated plugin architecture, separating each platform implementation into a dedicated packages. This restructuring improves long-term maintainability and allows each platform to evolve independently. These changes happened under the hood and requires no migration from the users.

Reviewer 2: ```suggestion
As part of an ongoing committment to code quality, the plugin has been reorganized into a fully federated plugin architecture, separating each platform implementation into a dedicated packages. This restructuring improves long-term maintainability and allows each platform to evolve independently. These changes happened under the hood and require no migration from users.
```

**Experimental desktop support**

Experimental support for Windows and macOS has been introduced via a WebView–based implementation using MapLibre GL JS. This is a simple solution to bring performant vector map rendering to these desktop platforms while broader native support continues to evolve.

<div 
  style={{ 
    marginBottom: "2rem", 
    width: "100%", 
    display: "flex", 
    flexDirection: "column", 
    alignItems: "center" 
  }}
>
  <Image
    src={flutteWindows}
    alt="MapLibre GL JS map in a Flutter App on Windows"
    style={{
      width: "100%",
      maxWidth: "650px", 
      height: "auto",
      display: "block",
      borderRadius: "8px",
      boxShadow: "0 4px 12px rgba(0,0,0,0.1)"
    }}
    loading="lazy"
    decoding="async"
  />

  <p
    style={{
      marginTop: "0.75rem",
      marginBottom: 0,
      lineHeight: "1.4",
      textAlign: "center",
      width: "100%",
      maxWidth: "550px", 
    }}
  >
    <em
      style={{
        fontSize: "0.85rem",
        color: "#666",
        fontStyle: "italic",
      }}
    >
      MapLibre GL JS map rendering within a Flutter App on Windows.
    </em>
  </p>
</div>
**Documentation and Future Roadmap**

> Alongside these changes, the documentation website has been migrated to Zensical to better align with the design of other MapLibre documentation: https://flutter-maplibre.pages.dev/docs/

Reviewer 2: ```suggestion
Aside from these changes, our [documentation website](https://flutter-maplibre.pages.dev/docs/) has been migrated to Zensical as well.
```

**We're looking for contributors.** Skills are plain markdown files, easy to read, review. We welcome your expansions, corrections and questions. If you've tackled geocoding, routing, styling, framework integration (React, Vue, Angular, Flutter, Swift), or performance — your experience can make a big impact.

See [contributing guidelines](https://github.com/maplibre/maplibre-agent-skills/blob/main/CONTRIBUTING.md) for how to write a skill, or [open an issue](https://github.com/maplibre/maplibre-agent-skills/issues) to propose a new one we haven't thought of yet.

## 🎨 Style Spec

MapLibre Style Spec continues to evolve to support more complex styling needs:
`split` and `join`, which allow better string manipulation, are now part of the spec and are expected to be incorporated in maplibre-gl-js soon.
`resampling` was also added to support other layers besides raster.
`symbol-elevation` will probably land soon to allow better support for placing symbols in a 3D map, check it out here:
https://github.com/maplibre/maplibre-style-spec/pull/1508

## 🦀 maplibre-rs

> During February focus has been on text rendering, with the following enhancements merged in:

Reviewer 2: ```suggestion
In February our focus has been on text rendering, with the following enhancements merged in:
```

    display: "flex",
    flexDirection: "column",
    alignItems: "center"
  }}

>

<Image
  src={maplibreRsText}
  alt="SDF text rendering in maplibre-rs"
  style={{
    width: "100%",
    maxWidth: "650px",
    height: "auto",
    display: "block",
    borderRadius: "8px",
    boxShadow: "0 4px 12px rgba(0,0,0,0.1)",
  }}
  loading="lazy"
  decoding="async"
/>

  <p
    style={{
      marginTop: "0.75rem",
      marginBottom: 0,
      lineHeight: "1.4",
      textAlign: "center",
      width: "100%",
      maxWidth: "550px", // 
    }}
  >
    <em
      style={{
        fontSize: "0.85rem",
        color: "#666",
        fontStyle: "italic",
      }}
    >
      Demonstration of the new SDF-based text rendering in maplibre-rs.
    </em>
  </p>
</div>

## 🧩 MapLibre Tile (MLT)

There is a lot of growing interest in [the new format](https://maplibre.org/news/2026-01-23-mlt-release/). Our current priority includes:

- We are rebuilding the test infrastructure to allow high quality decoders in all the major languages.
- There is also ongoing work on encoders in Rust and C++ to allow seamless integration into all kinds of tools including databases and tiles servers.

> - In MapLibre GL JS, we are working to leverage the native strengths of the MapLibre Tile format directly, moving away from the current method of converting data into an MVT-style model.

Reviewer 2: ```suggestion
- Leverage the native strengths of the MLT directly. For example, in MapLibre GL JS we aim to move away from the current method of converting data into an MVT-style model.
```

The main focus right now is quality, ensuring that decoding is handled perfectly across every supported language.

📣 Call for Volunteers: MLT Logo

> MLT project is looking for a volunteer to design our official logo. This is a great opportunity to contribute to the project and leave a lasting mark on the MapLibre ecosystem.

Reviewer 2: ```suggestion
The MLT project is looking for a volunteer to design our official logo. This is a great opportunity to contribute to the project and leave a lasting mark on the MapLibre ecosystem.
```

**Experimental desktop support**

Experimental support for Windows and macOS has been introduced via a WebView–based implementation using MapLibre GL JS. This is a simple solution to bring performant vector map rendering to these desktop platforms while broader native support continues to evolve.

<div 
  style={{ 
    marginBottom: "2rem", 
    width: "100%", 
    display: "flex", 
    flexDirection: "column", 
    alignItems: "center" 
  }}
>
  <Image
    src={flutteWindows}
    alt="MapLibre GL JS map in a Flutter App on Windows"
    style={{
      width: "100%",
      maxWidth: "650px", 
      height: "auto",
      display: "block",
      borderRadius: "8px",
      boxShadow: "0 4px 12px rgba(0,0,0,0.1)"
    }}
    loading="lazy"
    decoding="async"
  />

  <p
    style={{
      marginTop: "0.75rem",
      marginBottom: 0,
      lineHeight: "1.4",
      textAlign: "center",
      width: "100%",
      maxWidth: "550px", 
    }}
  >
    <em
      style={{
        fontSize: "0.85rem",
        color: "#666",
        fontStyle: "italic",
      }}
    >
      MapLibre GL JS map rendering within a Flutter App on Windows.
    </em>
  </p>
</div>
**Documentation and Future Roadmap**

> Alongside these changes, the documentation website has been migrated to Zensical to better align with the design of other MapLibre documentation: https://flutter-maplibre.pages.dev/docs/

Reviewer: ```suggestion
The documentation website has been migrated to Zensical alongside the broader MapLibre migration. [Check it out](https://flutter-maplibre.pages.dev/docs/) and let us know what you think.
```
